R/DataLoadingSaving.R

```diff
@@ -155,101 +147,86 @@ getDbCohortMethodData <- function(connectionDetails,
-    remove_duplicate_subjects = removeDuplicateSubjects,
-    washout_period = washoutPeriod,
-    restrict_to_common_period = restrictToCommonPeriod
-  )
-  DatabaseConnector::executeSql(connection, renderedSql)
-
-  sampled <- FALSE
-  if (maxCohortSize != 0) {
-    renderedSql <- SqlRender::loadRenderTranslateSql("CountCohorts.sql",
+  renderedSql <-
+    SqlRender::loadRenderTranslateSql(
+      "CreateCohorts.sql",
       packageName = "CohortMethod",
       dbms = connectionDetails$dbms,
       tempEmulationSchema = tempEmulationSchema,
-      target_id = targetId
+      cdm_database_schema = cdmDatabaseSchema,
+      exposure_database_schema = exposureDatabaseSchema,
+      exposure_table = exposureTable,
+      target_id = targetId,
+      comparator_id = comparatorId,
+      study_start_date = studyStartDate,
+      study_end_date = studyEndDate,
+      first_only = firstExposureOnly,
+      remove_duplicate_subjects = removeDuplicateSubjects,
+      washout_period = washoutPeriod,
+      restrict_to_common_period = restrictToCommonPeriod
     )
-    counts <- DatabaseConnector::querySql(connection, renderedSql, snakeCaseToCamelCase = TRUE)
-    ParallelLogger::logDebug("Pre-sample total row count is ", sum(counts$rowCount))
-    preSampleCounts <- dplyr::tibble(dummy = 0)
-    idx <- which(counts$treatment == 1)
-    if (length(idx) == 0) {
-      preSampleCounts$targetPersons <- 0
-      preSampleCounts$targetExposures <- 0
-    } else {
-      preSampleCounts$targetPersons <- counts$personCount[idx]
-      preSampleCounts$targetExposures <- counts$rowCount[idx]
-    }
-    idx <- which(counts$treatment == 0)
-    if (length(idx) == 0) {
-      preSampleCounts$comparatorPersons <- 0
-      preSampleCounts$comparatorExposures <- 0
-    } else {
-      preSampleCounts$comparatorPersons <- counts$personCount[idx]
-      preSampleCounts$comparatorExposures <- counts$rowCount[idx]
-    }
-    preSampleCounts$dummy <- NULL
-    if (preSampleCounts$targetExposures > maxCohortSize) {
-      message("Downsampling target cohort from ", preSampleCounts$targetExposures, " to ", maxCohortSize)
-      sampled <- TRUE
-    }
-    if (preSampleCounts$comparatorExposures > maxCohortSize) {
-      message("Downsampling comparator cohort from ", preSampleCounts$comparatorExposures, " to ", maxCohortSize)
-      sampled <- TRUE
-    }
-    if (sampled) {
-      renderedSql <- SqlRender::loadRenderTranslateSql("SampleCohorts.sql",
-        packageName = "CohortMethod",
-        dbms = connectionDetails$dbms,
-        tempEmulationSchema = tempEmulationSchema,
-        max_cohort_size = maxCohortSize
-      )
-      DatabaseConnector::executeSql(connection, renderedSql)
-    }
+  DatabaseConnector::executeSql(connection, renderedSql)
+
+  sampled <- FALSE
+  if (maxCohortSize != 0) {
+    outList <-
+      downSample(connection,
+                 tempEmulationSchema,
+                 targetId,
+                 maxCohortSize,
+                 sampled)
```

Comment:
Why does `downSample()` need `sampled`, which is guaranteed to be `FALSE`? 

Also, shouldn't `downSample()` need `comparatorId`?

Finally: If a function has more than 2 arguments I really prefer to use named arguments, even if it looks dumb, just to avoid correctly assigning the wrong value to the wrong parameter. So `connection = connection, `, etc.